src/sinks/splunk_hec/common/service.rs

```diff
@@ -4,32 +4,83 @@ use std::{
 };
 
 use crate::sinks::{
-    splunk_hec::common::{build_request, request::HecRequest, response::HecResponse},
+    splunk_hec::common::{build_uri, request::HecRequest, response::HecResponse},
     util::http::HttpBatchService,
 };
 use futures_util::future::BoxFuture;
 use http::Request;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+use tokio::sync::{
+    mpsc::{self, UnboundedSender},
+    oneshot::{self, Sender},
+};
 use tower::{Service, ServiceExt};
+use uuid::Uuid;
 use vector_core::event::EventStatus;
 
+use crate::sinks::UriParseError;
 use crate::{http::HttpClient, sinks::util::Compression};
 
+use super::acknowledgements::{run_acknowledgements, HecClientAcknowledgementsConfig};
+
+// A Splunk channel must be a GUID/UUID formatted value
+// https://docs.splunk.com/Documentation/Splunk/8.2.3/Data/AboutHECIDXAck#About_channels_and_sending_data
+lazy_static::lazy_static! {
+    static ref SPLUNK_CHANNEL: String = {
+        let mut buf = Uuid::encode_buffer();
+        Uuid::new_v4().to_hyphenated().encode_lower(&mut buf).to_string()
+    };
+}
```

Comment:
Does this need to be global to the entirety of Vector, or could it be a data member in `HecService`? Also, since it's a fixed-length string, could this be the same `[u8; 45]` as the `encode_buffer` returns?

Reply:
I'll refactor creating this logic into `HecService`. I forgot my original intentions with this method -- the purpose may have been lost after some refactoring.